### PR TITLE
Add keyword arguments to Pkg.checkout documentation

### PR DESCRIPTION
**Pull request by crbinz**
Addresses #13723. I did not modify packages.rst, because these seem like "advanced" options to me, and I can't think of a good example situation in which the user would want to change the defaults.

Also, this is my first PR to julia (and really, to any very large project), so if anything is wrong here, please correct me.

### CODE REVIEW

**Comment by tkelman** [comment]
Thanks! I would say "if `pull == true`".

Also documentation changes to docstrings need to be mirrored in the .jl source, see https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#improving-documentation

**Comment by crbinz** [comment]
@tkelman thanks for the feedback. I think I did what is required now, that paragraph is a little confusing to me.

File: base/docs/helpdb.jl [hunk 1]
```diff
@@ -11455,9 +11455,9 @@ If `pkg` has a URL registered in `Pkg.dir("METADATA")`, clone it from that URL o
 Pkg.clone(pkg)
 
 doc"""
-    checkout(pkg, [branch="master"])
+    checkout(pkg, [branch="master"]; merge=true, pull=true)
 
-Checkout the `Pkg.dir(pkg)` repo to the branch `branch`. Defaults to checking out the "master" branch. To go back to using the newest compatible released version, use `Pkg.free(pkg)`
+Checkout the `Pkg.dir(pkg)` repo to the branch `branch`. Defaults to checking out the "master" branch. To go back to using the newest compatible released version, use `Pkg.free(pkg)`\ . Changes are merged (fast-forward only) if the keyword argument ``merge == true``, and the latest version is pulled from the upsream repo if ``pull == true``\ .
```
Comment (tkelman) [inline base/docs/helpdb.jl]: Note that in the julia source the docstring is formatted as markdown by default, it's converted to RST by the script `doc/genstdlib.jl` which you'd need to run with a just-built julia.

So the backslash-space before period is not necessary in this copy, and the code formatting should be single backticks here. Running the genstdlib.jl script should result in this converting over to be equivalent to what you've written in the RST.

**Comment by crbinz** [comment]
Ok, trying to build julia now. So, the steps are: 1) edit the signature in pkg.rst, but not the body; 2) edit the signature and body in helpdb.jl; 3) run `doc/genstdlib.jl`. Is that right?

**Comment by tkelman** [comment]
You don't have to build Julia if you don't want to, it takes a while the first time. Just update the .jl version to be in markdown format, so remove the `\ ` and replace double backticks with single, then it should be fine.

**Comment by kshyatt** [comment]
@crbinz welcome to Julia! Yaaaaaay first time contributors :).

**Comment by crbinz** [comment]
@tkelman oh, good!

@kshyatt thanks! I've been a user for a year and a half or so now, so it feels good to make a (tiny) contribution.

**Comment by kshyatt** [comment]
@crbinz cents add up to dollars, small doc changes add a lot of value for the whole community :D

**Comment by tkelman** [comment]
Especially anywhere that the process may have confused you, we know how it works better than you do, so if our explanations clear up any details that didn't make sense at first and you have any suggestions for how to improve it, let us know. The docsystem should eventually be pure-Julia, pure-Markdown but it's going to take a lot of effort to get there, we're in a confusing in-between state right now.

**Comment by crbinz** [comment]
Re-reading that section in `CONTRIBUTING`, it does make sense to me now. I just needed to think through it a little more.

**Comment by hayd** [comment]
As an aside: shouldn't this kwarg be "fetch" rather than pull? It seems strange that you can have `pull=true` but `merge=false`... since a "pull" is a fetch followed by a merge.

**Comment by tkelman** [comment]
@hayd good call, though from [the code](https://github.com/JuliaLang/julia/blob/c7131061085bbc63d9bf5d4ae5b9458f63a72602/base/pkg/entry.jl#L231-L236) it looks like if `merge` and `pull` are both true then it ends up merging, then fetching, then merging again. Perhaps changing the `pull` kwarg to `fetch` would simplify things, as a separate change.

**Comment by hayd** [comment]
@tkelman should pull kwargs be deprecated or just :fire: ? (I'm not sure how to deprecate a kwarg)

**Comment by tkelman** [comment]
would have to be manually deprecated. not surr how widely used it is.

**Pull request merged**